src/rp2040/atomic-critical_section.adb

```diff
@@ -0,0 +1,94 @@
+with System.Machine_Code; use System.Machine_Code;
+
+package body Atomic.Critical_Section is
+
+   SPINLOCK31_Address : constant := 16#D000_017C#;
+
+   -----------------------------------
+   -- Local Subprogram Declarations --
+   -----------------------------------
+
+   procedure Spinlock_Lock
+     with Inline_Always;
+
+   procedure Spinlock_Unlock
+     with Inline_Always;
+
+   -------------------
+   -- Spinlock_Lock --
+   -------------------
+
+   procedure Spinlock_Lock is
+      use type Interfaces.Unsigned_32;
+
+      SPINLOCK31 : Interfaces.Unsigned_32 with
```

Comment:
You can define the spinlock only once at the top of the package.

Reply:
I've moved the definition to package scope in commit 839467f